### PR TITLE
Implemented logic to show product variants on the basis of the selected attribute(#1xc2eqg)

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/views/product-details.vue [hunk 1]
```diff
@@ -264,10 +268,37 @@ export default defineComponent({
...
       jobTotal: 'job/getTotal',
       userProfile: 'user/getUserProfile',
-      selectedBrand: 'user/getSelectedBrand'
-    })
+      selectedBrand: 'user/getSelectedBrand',
+    }),
+    filteredProducts () {
+      const filteredProducts = JSON.parse(JSON.stringify(this.current));
+      if(this.filter.size.length || this.filter.color.length){
+        filteredProducts.list.items = [];
```
Comment (adityasharma7) [inline src/views/product-details.vue:276]: We could remove this.

File: src/views/product-details.vue [hunk 2]
```diff
@@ -264,10 +268,37 @@ export default defineComponent({
...
+          })
+          if (hasSize || hasColor) return item;
+          else return null
+        }).filter((product: any) => product);
+      }
+      return filteredProducts;
+    }
   },
   methods: {
+    filterVarients (featureValue: any, type: any) {
```
Comment (adityasharma7) [inline src/views/product-details.vue:293]:
```suggestion
    filterVariants (featureValue: any, type: any) {
```

File: src/views/product-details.vue [hunk 3]
```diff
@@ -264,10 +268,37 @@ export default defineComponent({
...
+      if(this.filter[type].includes(featureValue)){
```
Comment (adityasharma7) [inline src/views/product-details.vue:294]: Fix spacing issues with if and else both

File: src/views/product-details.vue [hunk 4]
```diff
@@ -253,7 +253,11 @@ export default defineComponent({
       selectedVariants: {} as any,
       cusotmerLoyaltyOptions : JSON.parse(process.env?.VUE_APP_CUST_LOYALTY_OPTIONS),
       cusotmerLoyalty: '',
-      hasPromisedDate: true
+      hasPromisedDate: true,
+      filter:{
```
Comment (adityasharma7) [inline src/views/product-details.vue:257]:
```suggestion
      filters:{
```

File: src/views/product-details.vue (hunk 2 above)
Comment (adityasharma7) [reply src/views/product-details.vue:293]: Why not just filter?

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/product-details.vue (hunk 1 above)
Comment (disha1202) [reply src/views/product-details.vue]: Removed.

File: src/views/product-details.vue (hunk 2 above)
Comment (disha1202) [reply src/views/product-details.vue]: Previously there was a data property named filter. Updated the method name to filter and data property name to filters.

File: src/views/product-details.vue (hunk 3 above)
Comment (disha1202) [reply src/views/product-details.vue]: Fixed.

File: src/views/product-details.vue (hunk 4 above)
Comment (disha1202) [reply src/views/product-details.vue]: Done.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**